python/paddle/fluid/optimizer.py

```diff
@@ -1191,6 +1215,37 @@ def _dgc_op(self, param_var, clip_var, grad_var, u_var, v_var, k_var,
         dgc_op._set_attr(op_maker.kOpRoleVarAttrName(),
                          [param_var.name, grad_var.name])
 
+    def apply_gradients(self, params_grads):
```

Comment:
Is it better to modify the base class function, and add the DGC contition there:
```
not_dgc_params_grads = []
        dgc_params_grads = []
        for param, grad in params_grads:
            if not self._is_use_dgc(param, grad):
                not_dgc_params_grads.append((param, grad))
            else:
                dgc_params_grads.append((param, grad))

        # DGC clip and regularization in local
        not_dgc_params_grads = append_gradient_clip_ops(not_dgc_params_grads)

        # Add regularization if any
        not_dgc_params_grads = append_regularization_ops(not_dgc_params_grads,
                                                         self.regularization)
```